### PR TITLE
[No-1][Training]login-logout

### PR DESCRIPTION
**Pull request by loitq**
use laravel authentication.
Chạy qua phpcs, phpmd các file sau:

routes/web.php.
app/Http/Controllers

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -17,6 +17,7 @@ public function up()
             $table->bigIncrements('id');
             $table->string('name');
             $table->string('email')->unique();
+            $table->unsignedTinyInteger('role')->default(\App\User::USER); // default role amdin
```
Comment (hienl) [inline database/migrations/2014_10_12_000000_create_users_table.php]: I does not see the usage of https://laravel.com/docs/6.x/authentication "Database Considerations" yet

File: app/Http/Controllers/Admin/AdminController.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class AdminController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+
```
Comment (hienl) [inline app/Http/Controllers/Admin/AdminController.php]: add base condition Auth::check() , and Admin Role for display

File: app/Http/Controllers/User/BlogController.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+class BlogController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: function with only 1 return line no need this empty line

File: routes/web.php [hunk 4]
```diff
@@ -14,3 +14,27 @@
...
+        'namespace' => 'Admin'
+    ], function () {
+        // Controllers Within The "App\Http\Controllers\Admin" Namespace
+        Route::get('/', 'AdminController@index')->name('admin.index');
+    });
+
+    //Route User
+    Route::namespace('User')->group(function () {
+        // Controllers Within The "App\Http\Controllers\User" Namespace
+        Route::get('/', 'BlogController@index')->name('blog.index');
```
Comment (hienl) [inline routes/web.php:40]: Admin có quyền thêm xóa sửa blog , route như hiện tại admin ko access được blog page
> + Với tài khoản admin, bao gồm các quyền: thêm/xóa/sửa user, phân quyền cho user,
> thêm/xóa/sửa bài blog

File: webpack.mix.js [hunk 5]
```diff
@@ -12,4 +12,4 @@ const mix = require('laravel-mix');
  */
 
 mix.js('resources/js/app.js', 'public/js')
-    .sass('resources/sass/app.scss', 'public/css');
+   .sass('resources/sass/app.scss', 'public/css');
```
Comment (hienl) [inline webpack.mix.js:15]: why have this fixing ?

File: app/Http/Controllers/Auth/LoginController.php [hunk 6]
```diff
@@ -28,6 +30,22 @@ class LoginController extends Controller
      */
     protected $redirectTo = RouteServiceProvider::HOME;
 
+    /**
```
Comment (hienl) [inline app/Http/Controllers/Auth/LoginController.php:33]: Where is the usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=605672571 "Manually Authenticating Users"

File: resources/views/auth/login.blade.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                            </div>
+                        </div>
+
+                        <div class="form-group row">
+                            <div class="col-md-6 offset-md-4">
+                                <div class="form-check">
+                                    <input class="form-check-input" type="checkbox" name="remember" id="remember" {{ old('remember') ? 'checked' : '' }}>
+
+                                    <label class="form-check-label" for="remember">
+                                        {{ __('Remember Me') }}
```
Comment (hienl) [inline resources/views/auth/login.blade.php:48]: Why have remeber me function when in Sumary https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=605672571 said 'No'

File: app/Http/Controllers/Auth/LoginController.php [hunk 8]
```diff
@@ -28,6 +30,22 @@ class LoginController extends Controller
      */
     protected $redirectTo = RouteServiceProvider::HOME;
```
Comment (hienl) [inline app/Http/Controllers/Auth/LoginController.php:32]: Where is the usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=605672571 Other Authentication Methods

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 1 above)
Comment (hienl) [inline database/migrations/2014_10_12_000000_create_users_table.php]: comment and code does not match each other

File: database/seeds/UsersTableSeeder.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+            'email'    => "loitq@lifull-tech.vn",
+            'role'     => \App\User::ADMIN,
+            'password' => Hash::make("lifull@123"),
+        ]);
+
+        DB::table('users')->insert([
+            'name'     => "user",
+            'email'    => "user@lifull-tech.vn",
+            'password' => Hash::make("lifull@123"),
+            'role'     => \App\User::USER,
```
Comment (hienl) [inline database/seeds/UsersTableSeeder.php:22]: why each record have different order of fields ?

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 10]
```diff
@@ -17,6 +17,7 @@ public function up()
             $table->bigIncrements('id');
```
Comment (hienl) [inline database/migrations/2014_10_12_000000_create_users_table.php:17]: Where is usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=605672571 Logging out

File: routes/web.php [hunk 11]
```diff
@@ -14,3 +14,27 @@
 Route::get('/', function () {
     return view('welcome');
 });
+
+Auth::routes();
```
Comment (hienl) [inline routes/web.php:18]: Where is the usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=339395368 Route Parameters

File: resources/views/auth/register.blade.php [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+                                <input id="password-confirm" type="password" class="form-control" name="password_confirmation" required autocomplete="new-password">
+                            </div>
+                        </div>
+
+                        <div class="form-group row mb-0">
+                            <div class="col-md-6 offset-md-4">
+                                <button type="submit" class="btn btn-primary">
+                                    {{ __('Register') }}
+                                </button>
+                            </div>
```
Comment (hienl) [inline resources/views/auth/register.blade.php:69]: Where is usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=339395368 Form Method Spoofing

File: routes/web.php [hunk 13]
```diff
@@ -14,3 +14,27 @@
...
+
+Route::get('/home', 'HomeController@index')->name('home');
+
```
Comment (hienl) [inline routes/web.php:21]: Where is the usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=339395368 Accessing The Current Route

File: app/Http/Controllers/Auth/LoginController.php (hunk 8 above)
Comment (hienl) [inline app/Http/Controllers/Auth/LoginController.php:32]: Where is the usage of https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=0 Email Verification ?

**Review by hienl — CHANGES_REQUESTED** [review]
Please Fix and Confirm

File: webpack.mix.js (hunk 5 above)
Comment (loitq) [reply webpack.mix.js:15]: i run conmand ```npm run dev``` it recreated  file

File: app/Http/Controllers/Admin/AdminController.php (hunk 2 above)
Comment (loitq) [reply app/Http/Controllers/Admin/AdminController.php]: I used middleware auth... i think don't add Auth::check().

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 10 above)
Comment (loitq) [reply database/migrations/2014_10_12_000000_create_users_table.php:17]: feature logout has used in ruotes/web ``` Auth::routes() ```. 
it has written in file Illuminate\Foundation\Auth\AuthenticatesUsers.

File: routes/web.php (hunk 11 above)
Comment (loitq) [reply routes/web.php:18]: I thought it would be in feature edit blog or edit user.

File: resources/views/auth/register.blade.php (hunk 12 above)
Comment (loitq) [reply resources/views/auth/register.blade.php:69]: i think it would be in fearture add user or blogs

File: routes/web.php (hunk 13 above)
Comment (loitq) [reply routes/web.php:21]: i use for feature sidebar.

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 1 above)
Comment (loitq) [reply database/migrations/2014_10_12_000000_create_users_table.php]: I fixed in summary doccument 
https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=605672571

File: app/Http/Controllers/Auth/LoginController.php (hunk 8 above)
Comment (loitq) [reply app/Http/Controllers/Auth/LoginController.php:32]: i fixed in summary doccument. 
https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=0

File: app/Http/Controllers/Auth/LoginController.php (hunk 6 above)
Comment (loitq) [reply app/Http/Controllers/Auth/LoginController.php:33]: laravel support in folder /vendor/lluminate/Foundation/Auth/AuthenticatesUsers;

File: app/Http/Controllers/Auth/LoginController.php (hunk 8 above)
Comment (loitq) [reply app/Http/Controllers/Auth/LoginController.php:32]: i fixed in summary doccumemt. 
https://docs.google.com/spreadsheets/d/1bqeqmlFx-0867eX4onaM6IZYLpkewONk3wI8i1vICTs/edit#gid=605672571

**Review by hienl — APPROVED** [review]
LGTM

**Pull request merged**